pom.xml

```diff
@@ -278,7 +278,7 @@
         <storm.kafka.artifact.id>kafka_2.10</storm.kafka.artifact.id>
 
         <!-- kafka version used by new storm-kafka-client spout code -->
-        <storm.kafka.client.version>0.10.0.0</storm.kafka.client.version>
+        <storm.kafka.client.version>0.11.0.2</storm.kafka.client.version>
```

Comment:
We need to upgrade to at least 0.10.1.0, since that's where the `beginningOffsets` and `endOffsets` methods were added to the KafkaConsumer. I think upgrading the consumer and adding this patch would probably be a breaking change, because the spout won't work with 0.10.0.x client libraries/brokers anymore.

Reply:
yes, we need at-least 0.10.1.0 kafka clients and its a breaking change. We check for any other options

Reply:
Actually we are "effectively" dropping support of Kafka 0.9.x while moving from Storm 1.0.x to 1.1.0, because critical bugs in storm-kafka-client are not ported back to 1.0.x version line. I think we should try to avoid doing it again, but not sure for the cases like this. (bump the version for utilizing new feature)

Btw, if Kafka doesn't always guarantee forward compatibility, we may need to define support versions of Kafka (and others, if we could maintain) and maintain that.